src/model/enums/types.rs

```diff
@@ -0,0 +1,155 @@
+use serde::{Deserialize, Serialize};
+use strum::{AsRefStr, Display, EnumString};
+
+/// Album type - 'album', 'single', 'appears_on', 'compilation'
+#[derive(
+    Clone, Serialize, Deserialize, Copy, PartialEq, Eq, Debug, EnumString, AsRefStr, Display,
+)]
+#[serde(rename_all = "snake_case")]
+#[strum(serialize_all = "snake_case")]
+pub enum AlbumType {
+    Album,
+    Single,
+    AppearsOn,
+    Compilation,
+}
+
+///  Type: 'artist', 'album','track', 'playlist', 'show' or 'episode'
+#[derive(
+    Clone, Serialize, Deserialize, Copy, PartialEq, Eq, Debug, EnumString, AsRefStr, Display,
+)]
+#[serde(rename_all = "snake_case")]
+#[strum(serialize_all = "snake_case")]
+pub enum Type {
+    Artist,
+    Album,
+    Track,
+    Playlist,
+    User,
+    Show,
+    Episode,
+}
+
+/// additional_typs: track, episode
+#[derive(
+    Clone, Serialize, Deserialize, Copy, PartialEq, Eq, Debug, EnumString, AsRefStr, Display,
+)]
+#[serde(rename_all = "snake_case")]
+#[strum(serialize_all = "snake_case")]
+pub enum AdditionalType {
+    Track,
+    Episode,
+}
+
+/// currently_playing_type: track, episode, ad, unknown.
+#[derive(
+    Clone, Serialize, Deserialize, Copy, PartialEq, Eq, Debug, EnumString, AsRefStr, Display,
+)]
+#[serde(rename_all = "snake_case")]
+#[strum(serialize_all = "snake_case")]
+pub enum CurrentlyPlayingType {
+    Track,
+    Episode,
+    #[strum(serialize = "ad")]
```

Comment:
Why is this done with `strum` and not `serde`?

Reply:
Yes, it's a problem, I'll fix it.